docs/pipelines.md

```diff
@@ -274,9 +274,70 @@ spec:
 #### Param enum
 > :seedling: **Specifying `enum` is an [alpha](additional-configs.md#alpha-features) feature.** The `enable-param-enum` feature flag must be set to `"true"` to enable this feature.
 
-> :seedling: This feature is WIP and not yet supported/implemented. Documentation to be completed.
+Parameter declarations can include `enum` which is a predefine set of valid values that can be accepted by the `Pipeline` `Param`. For example, the valid/allowed values for `Param` "message" is bounded to `v1` and `v2`:
 
-Parameter declarations can include `enum` which is a predefine set of valid values that can be accepted by the `Pipeline`.
+``` yaml
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: pipeline-param-enum
+spec:
+  params:
+  - name: message
+    enum: ["v1", "v2"]
+    default: "v1"
+  tasks:
+  - name: task1
+    params:
+      - name: message
+        value: $(params.message)
+    steps:
+    - name: build
+      image: bash:3.2
+      script: |
+        echo "$(params.message)"
+```
+
+If the `Param` value passed in by `PipelineRun` is **NOT** in the predefined `enum` list, the `PipelineRun` will fail with reason `InvalidParamValue`.
+
+If a `PipelineTask` references a `Task` with `enum`, Tekton validates the **intersection** of enum specified in the referenced `Task` and the enum specified in the Pipeline `spec.params`. In the example below, the referenced `Task` accepts `v1` and `v2` as valid values, and the `Pipeline` accepts `v2` and `v3` as valid values. Only passing `v2` in the `PipelineRun` will lead to a sucessful execution.
```

Comment:
apologies for coming to this late, i wasn't involved in the design discussions

i find this behavior to be confusing -- for example, as a user of a pipeline, I'd be very confused why a pipelinerun failed when `v3` is listed as a valid value for the pipeline

what if we require that if both pipeline and task specify an enum, then the enum in the pipeline must be a subset of the enum in the task? that way, the issue is with the pipeline specification and not an issue that users run into occasionally even when they pass in certain values in the pipeline param enum

Reply:
Thanks @jerop for the suggestion. I've opened a TEP update to reflect what you suggested: https://github.com/tektoncd/community/pull/1106

Reply:
/hold cancel
https://github.com/tektoncd/community/pull/1106 is merged.

Based on the discussion in the TEP, I have added the subset validation, PTAL!